### PR TITLE
feat(authz): add opa bundle support

### PR DESCRIPTION
**Pull request by markphelps**
This adds the ability to use  [OPA bundles](https://www.openpolicyagent.org/docs/latest/management-bundles/) for loading in policies and data used to make authorization decisions

Currently, this PR adds the ability to pull a bundle from S3 (or Minio) via a similar configuration that we use for our [AWS S3 Storage](https://docs.flipt.io/configuration/storage#amazon-s3) for declarative state

It also allows advanced users to configure the Bundle API in OPA using the full configuration if they wish: https://www.openpolicyagent.org/docs/latest/management-bundles/#bundle-service-api

In the future we want to add 'nice' configuration to support authorization sources such as:

- GCP
- Azure
- OCI
- (eventually) Git

OPA already has examples/docs on how to use the bundle API for the first 3, however Git will likely need to be a custom implementation.

This PR also moves around the configuration a bit for our existing `local` authz source support which continues to use the lower-level rego SDK.

The plan is to extend the rego engine to also support pulling from Flipt Cloud API over HTTP, and likely to support Git as well since both of these methods will likely pull the 'raw' data and not a pre-packaged bundle like this new bundle engine does.

## TODO

- [x] ~~Add Integration Tests~~ PITA, Im resorting to testing manually
- [x] Fix linter issue
- [x] Fix tests
- [x] Test with S3
- [x] Test with Minio
- [x] Schema documentation
- [ ] Documentation
- [ ] (maybe) make space for HTTP policy/data source

### CODE REVIEW

File: internal/cmd/grpc.go [hunk 1]
```diff
@@ -228,11 +229,12 @@ func NewGRPCServer(
 		})),
 		grpc_prometheus.UnaryServerInterceptor,
 		otelgrpc.UnaryServerInterceptor(),
+		middlewaregrpc.ErrorUnaryInterceptor,
```
Comment (markphelps) [inline internal/cmd/grpc.go:232]: this ErrorInterceptor needs to come before authz/authn as the errors returned from those middleware should be caught by the error interceptor and interceptors work from outside in, so the first ones are on the outer layers

File: internal/config/authorization.go [hunk 2]
```diff
@@ -45,64 +55,201 @@ func (c *AuthorizationConfig) setDefaults(v *viper.Viper) error {
...
+			url.WriteString("https://s3")
+			if a.S3.Region != "" {
+				url.WriteString(".")
+				url.WriteString(a.S3.Region)
+			}
+			url.WriteString(".amazonaws.com")
+		}
+
+		tmpl = fmt.Sprintf(`
+services:
```
Comment (markphelps) [inline internal/config/authorization.go:220]: this is gross i'll admit, but its how the OPA SDK has to be configured https://github.com/open-policy-agent/opa/blob/f05497530d337dfd30dbd31851209d3a25c1cf95/sdk/options.go#L27

Comment (GeorgeMac) [reply internal/config/authorization.go:220]: I went looking to see if they at-least had a struct they use to parse it back out, so we could use that and marshal it before we pass it, but even theres is just a pile of json RawMessage:
https://github.com/open-policy-agent/opa/blob/f05497530d337dfd30dbd31851209d3a25c1cf95/config/config.go#L23-L48

File: internal/server/authz/engine/bundle/engine.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	if err != nil {
+		return false, err
+	}
+
+	allow, ok := dec.Result.(bool)
+	if !ok || !allow {
+		return false, nil
+	}
+
+	return true, nil
```
Comment (GeorgeMac) [inline internal/server/authz/engine/bundle/engine.go:89]: Take it or leave it:
```suggestion
	allow, _ := dec.Result.(bool)
	return allow, nil
```

**Review by GeorgeMac — APPROVED** [review]
Looks great. One take it or leave it 👍

**Pull request merged**